[WIP] Tutorial

vadimeisenbergibm opened the pull request:
The tutorial is different from the set of tasks/examples at istio.io in the following way:

1. Each task is self contained, has its setup/cleanup. The modules in the tutorial proceed step-by-step, without setup/cleanup, enabling quick demonstration of most of the features in one session. The participant perform set up of Bookinfo, the ingress gateway, the gateway for Istio dashboards, handles /etc/hosts/, and then follows the modules without the overhead of the setup. It uses multiple dashboards in each modules.
2. The tasks do not have any order. The modules in the tutorial flow in order, according to a developing story.
3. The tasks focus on a particular feature. The modules explain why the feature is needed and then present it. For example, dashboard, prometheus and tracing are shown in several modules, showing relevant features as they are required in the module.
4. The tasks explain the syntax of the Istio artifacts. The modules focus on the features and their usage, disregarding the syntax.
5. The tutorial is built to be followed by multiple participants simultaneously **on a single cluster**, each participant in a **separate namespace**. The tasks assume that the reader is the owner of the cluster and works in the default namespace. The tutorial can be set up by the cluster owner (a reader or a lab instructor), and then followed by the reader or by the instructor, or by multiple participants each in a separate namespace. The setup includes managing access permissions for the participants so they will not interfere with configurations of one another.
6. The tutorial presents Istio in the context of evolving from running a microservices application on Kubernetes, to injecting an Istio sidecar into a single microservices, then injecting Istio sidecars into all the microservices. It also presents some Kubernetes features (Kubernetes services and load balancing, Liveness Probes, Network Policies) during the course, so it is not only about Istio, but also about Kubernetes, Kubernetes + Istio.

I taught this tutorial in a class with 17 participants, it took about 8 hours net time. Currently it contains 26 modules, not counting introduction/setup/cleanup/summary. It can be previewed [here]( https://deploy-preview-1035--preliminary-istio.netlify.com/docs/tutorial/).

christian-posta commented:
So I think this is great stuff! This is straight forward enough and directly works with bookinfo that i think moving it to istio/contrib or somewhere in the istio repo makes sense. I'd be happy to contribute to it also.

One observation: On this step (https://github.com/vadimeisenbergibm/introduction-to-microservices-with-istio-bookinfo/tree/master/modules/06-add-istio-and-redeploy-bookinfo) we undeploy bookinfo and redeploy with istio Wonder if there's an opportunity to show how to gradually phase in istio to an existing application?

vadimeisenbergibm commented:
@christian-posta 
>  if there's an opportunity to show how to gradually phase in istio to an existing application?

We should weigh the tradeoff of making the tutorial more complicated vs. showing more features. I presented this tutorial yesterday, and this was the question I was asked at this step - do we have to redeploy the whole Bookinfo app? Can we do it gradually? The tutorial took two hours, with discussions and questions at each step.

I think it is actually a good idea, thanks! In the modules 07-09 and 11, it seems only productpage should be Istio enabled to route to reviews v1, v2 and v3. I will rewrite the module 06 to redeploy only the productpage.

vadimeisenbergibm commented:
@christian-posta and we also can consider to add an Advanced Tutorial section, to add more modules to demonstrate more Istio features in the same style. You are very welcome to contribute.

linsun commented:
@christian-posta @vadimeisenbergibm I agree, that is a great idea to show how to move some services into the mesh while the others are outside of the mesh.  I would be fine to have this in another tutorial tho so this one focus on introduce the value of Istio.

@vadimeisenbergibm really good work here!

geeknoid commented:
This is very nice.

What would you think about making this a top-level section in our docs? That is, adding a Tutorial section as a sibling of Tasks and Guides? Each module could be a separate page, and the YAML files would be in istio/istio/tutorial

vadimeisenbergibm commented:
> What would you think about making this a top-level section in our docs? That is, adding a Tutorial section as a sibling of Tasks and Guides? Each module could be a separate page, and the YAML files would be in istio/istio/tutorial
@geeknoid Sounds perfect! Let me do it.

geeknoid commented:
Let me know if you need help to add the Tutorial section. It should mostly
'just work' just by adding the directory under _docs.


On Tue, Feb 27, 2018 at 8:51 AM Vadim Eisenberg <notifications@github.com>
wrote:

> What would you think about making this a top-level section in our docs?
> That is, adding a Tutorial section as a sibling of Tasks and Guides? Each
> module could be a separate page, and the YAML files would be in
> istio/istio/tutorial
> @geeknoid <https://github.com/geeknoid> Sounds perfect! Let me do it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/istio.github.io/pull/1035#issuecomment-368945911>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AVucHSMYKQusQ34UOGjIvQtcZ1K-ZFOhks5tZDJ5gaJpZM4STdGK>
> .
>

vadimeisenbergibm commented:
@geeknoid yes, will do, thanks.

geeknoid commented on _docs/tutorial/00-setup.md:
```diff
@@ -0,0 +1,26 @@
+---
+{% include home.html %}
+
+# Introduction to microservices with Bookinfo sample application, Kubernetes and Istio
```
Don't add any single # titles, the website infra automatically takes the title: and inserts a level 1 heading with it.

geeknoid commented on _docs/tutorial/00-setup.md:
````diff
@@ -0,0 +1,26 @@
+
+1. Download Istio and Bookinfo source:
+   ```bash
+   make setup
````
Maybe include the git clone + cd commands?

geeknoid commented on _docs/tutorial/00-setup.md:
````diff
@@ -0,0 +1,26 @@
+   ```bash
+   alias istioctl=$(pwd)/istio-*/bin/istioctl
+   ```
+2. Go over `modules`, by their prefix number, issuing `cd` to each modules's directory.
````
How about adding a Previous/Next links at the bottom of each page? Something like:

```
## What's next
* Next Module: <title of next module>
* Previous Module: <title of previous module>
```

geeknoid commented on _docs/tutorial/01-ratings-service.md:
````diff
@@ -0,0 +1,29 @@
+   npm start 9080
+   ```
+
+1. Access http://localhost:9080/ratings/7
````
Here and elsewhere, maybe show a curl command?

geeknoid commented on _docs/tutorial/02-run-ratings-in-docker.md:
```diff
@@ -0,0 +1,37 @@
+---
+title: Run ratings in docker
```
Docker

geeknoid commented on _docs/tutorial/02-run-ratings-in-docker.md:
```diff
@@ -0,0 +1,37 @@
+---
+title: Run ratings in docker
+overview: Overview
```
Need an overview

geeknoid commented on _docs/tutorial/02-run-ratings-in-docker.md:
````diff
@@ -0,0 +1,37 @@
+1. Access http://localhost:9081/ratings/7
+
+1. Observe the running container
+   ```
````
Here and elsewhere, please use ```bash.

geeknoid commented on _docs/tutorial/03-run-bookinfo-with-kubernetes.md:
```diff
@@ -0,0 +1,37 @@
+---
+title: Run bookinfo with kubernetes
+overview: Overview
```
Needs an overview

geeknoid commented on _docs/tutorial/03-run-bookinfo-with-kubernetes.md:
```diff
@@ -0,0 +1,37 @@
+
+This learning module shows you an application of four microservices: _productpage_, _details_, _ratings_ and _reviews_. The application is called Bookinfo and is described [here](https://istio.io/docs/guides/bookinfo.html). Consider the application there as the final version, in which the _reviews_ microservice has three versions _v1_, _v2, _v3_. In this module we start with the application with  ...
+
+1. Skim `bookinfo.yaml` - this is a Kubernetes deployment spec of the app. Notice the services and the deployments, and also the replication: 3 replicas of each microservice.
```
bookinfo.yaml could be a link...

geeknoid commented on _docs/tutorial/02-run-ratings-in-docker.md:
```diff
@@ -0,0 +1,37 @@
+{% include home.html %}\n# Run the _ratings_ service in a docker container.
+This learning module shows how you can create a docker image and run it locally.
+
+1. Observe the `Dockerfile`
```
Make Dockerfile a link?

geeknoid commented on _docs/tutorial/03-run-bookinfo-with-kubernetes.md:
````diff
@@ -0,0 +1,37 @@
+   ```
+   kubectl get pods
+   ```
+1. Edit `ingress.yaml` - specify your host instead of `your host`.
````
Link for ingress.yaml?

geeknoid commented on _docs/tutorial/04-testing-in-production.md:
```diff
@@ -0,0 +1,33 @@
+---
+title: Testing in production
+overview: Overview
```
Needs an overview

geeknoid commented on _docs/tutorial/04-testing-in-production.md:
```diff
@@ -0,0 +1,33 @@
+---
+{% include home.html %}\n# Testing a microservice in "production"
+
+Let's perform some testing of our microservice . This learning module exemplifies that you can test your microservices in production!
```
Remove extra space are "microservice"

geeknoid commented on _docs/tutorial/04-testing-in-production.md:
```diff
@@ -0,0 +1,33 @@
+
+Let's perform some testing of our microservice . This learning module exemplifies that you can test your microservices in production!
+
+1. Let's send some requests to our microservice from inside the cluster, we will use a dummy pod, `sleep`.
```
Link to sleep.yaml?

geeknoid commented on _docs/tutorial/04-testing-in-production.md:
````diff
@@ -0,0 +1,33 @@
+   curl http://ratings:9080/ratings/7
+   ```
+1. Let's do some [chaos testing](http://www.boyter.org/2016/07/chaos-testing-engineering/) in production and see how our application reacts. After each chaos operation, access your `http://<your host>/productpage` and see if anything  changed. Also check the pods status with `kubectl get pods`.
+   1. Let's kill the _details_ service in one pod.
````
Here and elsewhere: kill -> terminate

ZackButcher replied:
![image](https://user-images.githubusercontent.com/1923135/36991270-04a462fc-205c-11e8-8a70-2016701987db.png)

geeknoid commented on _docs/tutorial/05-adding-a-new-version-of-a-microservice.md:
```diff
@@ -0,0 +1,71 @@
+---
+title: Adding a new version of a microservice
+overview: Overview
```
Needs an overview.

geeknoid commented on _docs/tutorial/05-adding-a-new-version-of-a-microservice.md:
```diff
@@ -0,0 +1,71 @@
+---
+{% include home.html %}\n# Deploy and release a new version of _reviews_
+
+Let's deploy a new version of the _reviews_ microservice, the one that will return the ratings provided by reviewers, as a number of stars, with the color of stars. In the real life, we would perform lint tests, unit tests, integration tests, end-to-end tests and tests in the staging environment.
```
in the real life -> In real life
in the staging -> in a staging

geeknoid commented on _docs/tutorial/05-adding-a-new-version-of-a-microservice.md:
````diff
@@ -0,0 +1,71 @@
+  kubectl apply -f bookinfo-reviews-v2-without-app-label.yaml
+  ```
+
+2. Let's access our application and see that the deployed microservice did not disrupt it. So far so good.
````
curl command to do this?

vadimeisenbergibm replied:
@geeknoid I do not think it is a good idea to access the productpage by curl. I do not see a reason why anyone would not use the browser. You just open the browser with the productpage once, and then hit refresh multiple times during the tutorial. This will show what happens graphically - black/red/no review stars, reviews unavailable etc. curl will just dump the html, without any clue what happened. During this tutorial I think it is important to see the productpage visually - to convince yourself that you can perform multiple transformations without disrupting the application.

vadimeisenbergibm replied:
@geeknoid And this is again an example that this is a part of a continuous tutorial - you open the browser at the step 3 to access the productpage, and then you just hit refresh on the following steps.

geeknoid commented on _docs/tutorial/05-adding-a-new-version-of-a-microservice.md:
````diff
@@ -0,0 +1,71 @@
+    ```bash
+    kubectl exec -it $(kubectl get pod -l app=sleep -o jsonpath='{.items[0].metadata.name}') -- curl $DETAILS_V2_POD_IP:9080/reviews/7
+    ```
+  3. Perform a primitive "load testing" - send requests for 10 times in a row:
````
Perform a primitive "load testing" -> Perform primitive load testing
requests for 10 -> requests 10

geeknoid commented on _docs/tutorial/05-adding-a-new-version-of-a-microservice.md:
````diff
@@ -0,0 +1,71 @@
+     kubectl exec -it $(kubectl get pod -l app=sleep -o jsonpath='{.items[0].metadata.name}') bash
+     for i in {1..10}; do curl -o /dev/null -s -w "%{http_code}\n" <the value of DETAILS_V2_POD_IP>:9080/reviews/7; done
+     ```
+4. Now we are rather confident that our new version of _reviews_ will work and we will release it. We will release a single replica of it into production - the real production traffic will arrive to our new version. In the current setting, 75% of the traffic will arrive to the old version (three pods of the old version) and 25% will arrive to the new version (one pod).
````
In the current -> With the current

geeknoid commented on _docs/tutorial/05-adding-a-new-version-of-a-microservice.md:
````diff
@@ -0,0 +1,71 @@
+  ```
+  kubectl get pods --show-labels -l version=v2
+  ```
+5. Now we access the application web page and see that the black stars appear for ratings! We will hit the page several times and see that the sometimes the page is returned with stars (approximately 25% of the time) and sometimes without stars (approximately 75% of the time)
````
that the sometimes -> that sometimes

geeknoid commented on _docs/tutorial/05-adding-a-new-version-of-a-microservice.md:
````diff
@@ -0,0 +1,71 @@
+  ```
+5. Now we access the application web page and see that the black stars appear for ratings! We will hit the page several times and see that the sometimes the page is returned with stars (approximately 25% of the time) and sometimes without stars (approximately 75% of the time)
+
+6. If we encounter any problems we can quickly undeploy the new version, so only the old version will be used.
````
Commands for doing so?

geeknoid commented on _docs/tutorial/05-adding-a-new-version-of-a-microservice.md:
```diff
@@ -0,0 +1,71 @@
+
+6. If we encounter any problems we can quickly undeploy the new version, so only the old version will be used.
+
+7. Next we will increase the replicas of our new version. We can do it gradually, carefully checking that our number of errors does not increase:
```
our number -> the number

geeknoid commented on _docs/tutorial/05-adding-a-new-version-of-a-microservice.md:
```diff
@@ -0,0 +1,71 @@
+
+We performed the update of _reviews_ pretty well. First, we deployed the new version without directing to it any production traffic. We tested it in the production environment, on test traffic. We checked that the new version does not produce any errors. We released the new version, gradually increasing the production traffic to it. Finally, we decommissioned the old version.
+
+It all went good, however we want to improve our release strategy. First, we want to allow our testers to test the new version end-to-end in production. For that we need an ability to drive the traffic to our new version by request parameters, for example by the user name stored in a cookie. Kubernetes cannot provide this routing functionality. Also, we would like to perform _shadowing_ of the pr ...
```
went good -> went well
Kubernetes cannot do it -> Kubernetes can't do that either.
fine grained -> fine-grained

geeknoid replied:
Maybe mention all three things, and then just say once "Kubernetes is unable to directly help with any of these tasks".

geeknoid commented on _docs/tutorial/05-adding-a-new-version-of-a-microservice.md:
```diff
@@ -0,0 +1,71 @@
+1. Implement the required functionality in the code. Most of the functionality is already implemented in various libraries, for example in the [Hystrix](https://github.com/Netflix/Hystrix) of Netflix for the Java programming language. However, now we have to change our code to call the functions from the libraries. We have to put additional effort, our code will bloat, business logic will be mixe ...
+2. Use a service mesh. In a service mesh, we put all the reporting, routing, policies, security logic in _sidecar_ proxies, injected into our pods *transparently* to our application. The business logic remains in the code of the application, no changes are required to the application code.
+
+Enters [Istio service mesh](istio.io). Istio can perform the tasks mentioned here and much more. In the next learning modules, we will explore various good features Istio can provide.
```
istio.io -> https://istio.io
various good features Istio can provide -> various features Istio provides

geeknoid commented on _docs/tutorial/06-add-istio.md:
```diff
@@ -0,0 +1,35 @@
+---
+title: Add istio
+overview: Overview
```
Needs an overview

geeknoid commented on _docs/tutorial/06-add-istio.md:
```diff
@@ -0,0 +1,35 @@
+layout: docs
+type: markdown
+---
+{% include home.html %}\n# In this module we will deploy the Istio control plane. We will enable Istio for a single microservice.
```
Remove \n# and add a linefeed.

geeknoid commented on _docs/tutorial/06-add-istio.md:
```diff
@@ -0,0 +1,35 @@
+---
+{% include home.html %}\n# In this module we will deploy the Istio control plane. We will enable Istio for a single microservice.
+
+As we saw in the previous module, Kubernetes does not provide us all the functionality we need for effective operating of our microservices. Istio comes to our help.
```
for effective operating of our -> to effectively operate our

geeknoid commented on _docs/tutorial/06-add-istio.md:
```diff
@@ -0,0 +1,35 @@
+
+First we deploy the _Istio control plane_. Then we enable Istio on a single microservice, _productpage_. The rest of the application will continue to operate as previously. Note that we can enable Istio gradually, microservice by microservice. Also note that Istio is enabled transparently to the microservices, we do not change the microservices code. And also note that we enable Istio without dis ...
+
+We use `istiocl kube-inject` command to inject Istio _sidecar proxies_ into the microservice pods.
```
use -> use the

geeknoid commented on _docs/tutorial/06-add-istio.md:
````diff
@@ -0,0 +1,35 @@
+   ```bash
+   kubectl apply -f ../../istio-*/install/kubernetes/istio.yaml
+   ```
+1. Verify that Istio is started correctly, all the pods in `istio-system` namespace are running.
````
is started -> started

geeknoid commented on _docs/tutorial/06-add-istio.md:
```diff
@@ -0,0 +1,35 @@
+
+1. Access the application and verify that the application continues to work. Note that Istio was added **transparently**, the code of the original application did not change.
+
+2. Check the pods of the _productpage_ and see that now each replica has two containers. The first container is the microservice itself, the second is the sidecar proxy attached to it.
```
Show commands and sample output?

vadimeisenbergibm replied:
@geeknoid I will add `kubectl get pods`. Regarding the output, I have my doubts. In this tutorial I do not show any output to make it as short as possible, only the instructions to run. Originally, I used this tutorial for my demonstration of Istio. So this tutorial is for quick self-learning or for a quick demonstration of most of the Istio features. Let's think if we want to expand the tutorial with the output.

vadimeisenbergibm replied:
@geeknoid I have second thoughts about the output, maybe I should add output as in tasks and blog posts.

geeknoid commented on _docs/tutorial/06-add-istio.md:
```diff
@@ -0,0 +1,35 @@
+
+2. Check the pods of the _productpage_ and see that now each replica has two containers. The first container is the microservice itself, the second is the sidecar proxy attached to it.
+
+3. Note that Kubernetes replaced the original pods of _productpage_ with the Istio-enabled pods, transparently and incrementally, performing what is called [rolling update](https://kubernetes.io/docs/tutorials/kubernetes-basics/update-intro/). Kubernetes terminated an old pod only when a new pod started to run, and it transparently switched the traffic to the new pods, one by one. (To be more pre ...
```
rolling -> a rolling

geeknoid commented on _docs/tutorial/07-canary-release-with-istio.md:
```diff
@@ -0,0 +1,40 @@
+---
+title: Canary release with istio
+overview: Overview
```
Needs an overview.

geeknoid commented on _docs/tutorial/07-canary-release-with-istio.md:
```diff
@@ -0,0 +1,40 @@
+layout: docs
+type: markdown
+---
+{% include home.html %}\n# Canary Release with Istio
```
Remove \n# and add a linefeed

geeknoid commented on _docs/tutorial/07-canary-release-with-istio.md:
```diff
@@ -0,0 +1,40 @@
+---
+{% include home.html %}\n# Canary Release with Istio
+In this learning module we will deploy a new version of the _reviews_ microservice again,
+this time with Istio enabled. We will release our new version to the `jason` user only (`jason` is our tester). It will allow the `jason` user to test the whole application end-to-end in production, with our new version.
```
Maybe use a name other than jason, since it gets confusing relative to json IMO.

vadimeisenbergibm replied:
@geeknoid I opened an issue https://github.com/istio/istio/issues/3945. Jason is used in Bookinfo sample rules, if we decide to change it we should rename it in istio/istio and istio/github.istio.io.

geeknoid commented on _docs/tutorial/07-canary-release-with-istio.md:
```diff
@@ -0,0 +1,40 @@
+In this learning module we will deploy a new version of the _reviews_ microservice again,
+this time with Istio enabled. We will release our new version to the `jason` user only (`jason` is our tester). It will allow the `jason` user to test the whole application end-to-end in production, with our new version.
+
+1. Let's specify a routing rule that all the production traffic will flow to the version _v1_ of all the microservices:
```
rule that all the production traffic will flow to the version _v1_-> rule to send all production traffic to version _v1_ of

geeknoid commented on _docs/tutorial/07-canary-release-with-istio.md:
````diff
@@ -0,0 +1,40 @@
+   kubectl apply -f  ../05-adding-a-new-version-of-a-microservice/bookinfo-reviews-v2-with-app-label.yaml
+   ```
+
+3. Let' access the application's web page multiple times and verify that our new version is not called.
````
Let's
Show curl command?

geeknoid commented on _docs/tutorial/07-canary-release-with-istio.md:
```diff
@@ -0,0 +1,40 @@
+
+3. Let' access the application's web page multiple times and verify that our new version is not called.
+
+4. Now, let's apply an Istio [route rule](https://istio.io/docs/reference/config/istio.routing.v1alpha1.html) to allow `jason` user to access our new version for testing:
```
Here and elsewhere, replace https://istio.io with {{home}}
allow -> allow the

geeknoid commented on _docs/tutorial/07-canary-release-with-istio.md:
````diff
@@ -0,0 +1,40 @@
+   istioctl create -f ../../istio-*/samples/bookinfo/kube/route-rule-reviews-test-v2.yaml
+   ```
+
+5. Let's login as `jason` (any password would do). We will see that now the reviews have black stars (our new version is used). Now we can let a human tester or a testing tool test our new version on the whole application
````
Missing period at end.

geeknoid commented on _docs/tutorial/08-enable-istio-on-all-the-microservices.md:
```diff
@@ -0,0 +1,51 @@
+---
+title: Enable istio on all the microservices
+overview: Overview
```
Here and elsewhere: needs an overview.

This overview is displayed when you hover over the title of the article in the left-hand nav area.

geeknoid commented on _docs/tutorial/08-enable-istio-on-all-the-microservices.md:
```diff
@@ -0,0 +1,51 @@
+layout: docs
+type: markdown
+---
+{% include home.html %}\n# In this module we will enable Istio on our whole application and on the Ingress.
```
Here and elsewhere, remove \n#

geeknoid commented on _docs/tutorial/08-enable-istio-on-all-the-microservices.md:
```diff
@@ -0,0 +1,51 @@
+
+Previously we deployed the Istio control plane and enabled Istio on a single microservice, _productpage_. We can proceed to enable Istio on the microservices incrementally, one by one, to get the functionality provided by Istio for more and more microservices. For the purpose of this tutorial, we will just enable Istio on the remaining microservices in one stroke. We will also enable Istio on our ...
+
+1. Redeploy Bookinfo application, Istio-enabled. _productpage_ will not be redeployed since it already has Istio injected, its pods will not be changed.
```
Redeploy -> Redeploy the

geeknoid commented on _docs/tutorial/08-enable-istio-on-all-the-microservices.md:
```diff
@@ -0,0 +1,51 @@
+
+1. Access the application after determining Ingress IP and port. Note that Istio was added **transparently**, the original application did not change. It was added on the fly, without the need to undeploy and redeploy the whole application, without hurting the application's availability.
+
+2. Check the application pods and verify that now each pod has two containers. One container is the microservice itself, the other is the sidecar proxy attached to it.
```
Show command with sample output?

ldemailly commented:
can you refactor rather than copy/clone the existing samples; it'll be a maintenance nightmare

vadimeisenbergibm commented:
@ldemailly I cannot refactor the existing tasks since the tasks serve the different purposes from the tutorial. In my opinion, both the tasks and the tutorial are needed.

I explained the differences in the first comment of the PR:
> The new tutorial is different from the set of tasks at istio.io in the following way:
> 
> Each task is self contained, has its setup/cleanup. The modules in the tutorial proceed step-by-step, without setup/cleanup, enabling quick demonstration of most of the basic features in one session.
The tasks do not have any order. The modules in the tutorial flow in order, according to a developing story.
The tasks focus on a particular feature. The modules explain why the feature is needed and then present it.
The tasks explain the syntax of the Istio artifacts. The modules focus on the features and their usage, disregarding the syntax.

ldemailly commented:
this looks great btw, what's the preview link ?

also what's the conclusion on blog vs tutorial vs tasks vs guides - I think we should not have a new top level but maybe we can rename guides to tutorials or some such

vadimeisenbergibm commented:
@ldemailly let me finalize it and publish a link. @geeknoid proposed to have a new top level - Tutorial.

I would like us to decide on the guidelines: Tasks vs. Guides vs. Blog vs. Tutorial. I do not know how we should proceed.

frankbu commented:
@ldemailly I don't think that we should merge guides and tutorials, since they serve different purposes. Guides are supposed to just be recommended "guided views" through tasks (i.e., do task a, then b, then c to understand some feature of Istio). This tutorial, on the other hand, is a complete standalone learning exercise.

ldemailly commented:
sgtm so do you want to move this out of a blog ?

I am assuming it would show up here on the left with others as new sub-toplevel under docs?

![screen shot 2018-03-05 at 12 12 47 pm](https://user-images.githubusercontent.com/3664595/36997473-95dbda2c-206e-11e8-9217-6c488e3c65b7.png)

vadimeisenbergibm commented:
@ldemailly @geeknoid @frankbu @christian-posta @linsun @ZackButcher please check _Tutorial_ at  https://vadimeisenbergibm.github.io/docs/

Your comments are welcome.

ldemailly commented on _docs/tutorial/00-setup.md:
```diff
@@ -0,0 +1,30 @@
+
+1. Install [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git), [curl](https://curl.haxx.se/download.html), [node.js](https://nodejs.org/en/download/), [Docker](https://docs.docker.com/install/)
+and get access to a [Kubernetes](https://kubernetes.io) cluster.
+For example, you can try the [IBM Cloud Container Service](https://console.bluemix.net/docs/containers/container_index.html#container_index).
```
should we put all or none of the providers ? or the main contributor's providers (ie both GCP and IBM)?

vadimeisenbergibm replied:
I propose that we start with this PR with IBM. Then we add all other providers as separate PRs, by the provider people. If you think we should add GCP right now, please submit a PR to my branch with GCP instructions.

ldemailly replied:
done. thx

ldemailly commented on _docs/tutorial/00-setup.md:
```diff
@@ -0,0 +1,30 @@
+
+1. This tutorial assumes that you perform the commands of the steps in the Istio directory that you downloaded and extracted in the steps 1 and 2 of the [Installation Steps of the Kubernetes Quick Start instructions]({{home}}/docs/setup/kubernetes/quick-start.html#installation-steps).
+
+1. Download Istio sources into the Istio directory. **Note** that Istio source code is under [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0) license.
```
Why are the sources needed? It should work with just the release tgz
the review of the sources could just be links into github

I don't think "git clone..." should be part of user facing instructions - it can only lead to issue with mismatch between release version and state of master etc


the install istio step for instance will not work from source

vadimeisenbergibm replied:
Good point. The sources are needed for the only one reason: to run _ratings_ as a node.js application, and as a Docker container. So basically, only https://github.com/istio/istio/tree/master/samples/bookinfo/src/ratings is needed. I will instruct the users to download the source code from https://github.com/istio/istio/releases, so the source code will always match the release.

ldemailly replied:
or you could just point them https://github.com/istio/istio/blob/master/samples/bookinfo/src/ratings/ratings.js as it's just one file ?
or wget -r https://github.com/istio/istio/blob/master/samples/bookinfo/src/ratings/
if you also want the Dockerfile etc (why would you ?)

vadimeisenbergibm replied:
I changed the setup to instruct to download the sources from https://github.com/istio/istio/releases, please check.
The sources are already packaged there, they match the release, no need to install `wget`.

I need the Docker file for the step 2, Running a single microservice in a Docker container.

ldemailly commented:
also please update your branch and run "rake test" / fix bad links (martin made using {{home}} instead of / mandatory for the https://archive.istio.io/ sites to work well. you can use relative links instead if you want)

geeknoid commented on _docs/tutorial/14-dashboard-with-istio.md:
````diff
@@ -0,0 +1,28 @@
+  kubectl -n istio-system port-forward $(kubectl -n istio-system get pod -l app=grafana -o jsonpath='{.items[0].metadata.name}') 3000:3000 &
+  ```
+
+3. Access the dashboard on local machine:  http://localhost:3000/dashboard/db/istio-dashboard
````
Make that a clickable link.

geeknoid commented on _docs/tutorial/15-distributed-tracing-with-istio.md:
```diff
@@ -0,0 +1,29 @@
+---
+title: Distributed tracing with Istio
+overview: Add Zipkin distributed trace system.
```
trace system -> tracing

geeknoid commented on _docs/tutorial/15-distributed-tracing-with-istio.md:
```diff
@@ -0,0 +1,29 @@
+---
+{% include home.html %}
+
+In this step, we will add [Zipkin distributed trace system](https://zipkin.io) as part of our Istio infrastructure.
```
trace system -> tracing

geeknoid commented on _docs/tutorial/15-distributed-tracing-with-istio.md:
````diff
@@ -0,0 +1,29 @@
+   ```bash
+   kubectl apply -f install/kubernetes/addons/zipkin.yaml
+   ```
+2. Check the pods at istio-system namespaces and wait for the pod of Zipkin to start running:
````
pod of Zipkin -> Zipkin's pod

geeknoid commented on _docs/tutorial/15-distributed-tracing-with-istio.md:
````diff
@@ -0,0 +1,29 @@
+   kubectl port-forward -n istio-system $(kubectl get pod -n istio-system -l app=zipkin -o jsonpath='{.items[0].metadata.name}') 9411:9411 &
+   ```
+
+3. Access the Zipkin dashboard on local machine: http://localhost:9411
````
Make the link clickable.

geeknoid commented on _docs/tutorial/16-generating-service-graph-with-istio.md:
````diff
@@ -0,0 +1,29 @@
+  ```bash
+  kubectl apply -f install/kubernetes/addons/servicegraph.yaml
+  ```
+2. Check the pods at istio-system namespaces and wait for the pod of Zipkin to start running:
````
pod of Zipkin -> Servicegraph's pod

geeknoid commented on _docs/tutorial/16-generating-service-graph-with-istio.md:
````diff
@@ -0,0 +1,29 @@
+
+   ```
+
+3. Access the generated service graph on the local machine:  http://localhost:8088/dotviz
````
Make link clickable.

geeknoid commented on _docs/tutorial/98-summary.md:
```diff
@@ -0,0 +1,27 @@
+---
+title: Summary
+overview: Tutorial summary and the next steps.
```
the next -> next

geeknoid commented on _docs/tutorial/99-cleanup.md:
````diff
@@ -0,0 +1,29 @@
+  ```
+
+## What's next
+{% if page.previous.url %}
````
No need for the conditional.

vadimeisenbergibm commented on _docs/tutorial/03-run-bookinfo-with-kubernetes.md:
```diff
@@ -0,0 +1,39 @@
+
+This step shows you an application of four microservices: _productpage_, _details_, _ratings_ and _reviews_. The application is called Bookinfo and is described [here]({{home}}/docs/guides/bookinfo.html). Consider the application there as the final version, in which the _reviews_ microservice has three versions _v1_, _v2, _v3_. In this step we start with the application with the first version of  ...
+
+1. Skim bookinfo.yaml TODO: ADD LINK - this is a Kubernetes deployment spec of the app. Notice the services and the deployments, and also the replication: 3 replicas of each microservice.
```
yep, these depend on https://github.com/istio/istio/pull/3940

ldemailly commented on _docs/tutorial/99-cleanup.md:
```diff
@@ -25,5 +25,5 @@ type: markdown
 ## What's next
 {% if page.previous.url %}
-Previous step: [{{page.previous.title}}]({{page.previous.url}})
+Previous step: [{{page.previous.title}}]({{home}}/{{page.previous.url}})
```
@geeknoid it seems like a bug in "page.previous.url" if it needs this prefix, no ?

vadimeisenbergibm replied:
@ldemailly @geeknoid `page.previous.url` and `page.next.url` are the Jekyll constructs. I saw it in http://david.elbe.me/jekyll/2015/06/20/how-to-link-to-next-and-previous-post-with-jekyll.html.

Adding `{{home}}` (without a slash) to `page.previous.url` and `page.next.url` works and the rake test is happy.

geeknoid commented:
I don't know where page.previous and page.next come from, I don't define
them in our stuff. So it's a baked-in Jekyll thing I guess. I was surprised
to see this in the first place.


On Tue, Mar 6, 2018 at 1:33 PM, Laurent Demailly <notifications@github.com>
wrote:

> *@ldemailly* commented on this pull request.
> ------------------------------
>
> In _docs/tutorial/99-cleanup.md
> <https://github.com/istio/istio.github.io/pull/1035#discussion_r172667246>
> :
>
> > @@ -25,5 +25,5 @@ type: markdown
>
>  ## What's next
>  {% if page.previous.url %}
> -Previous step: [{{page.previous.title}}]({{page.previous.url}})
> +Previous step: [{{page.previous.title}}]({{home}}/{{page.previous.url}})
>
> @geeknoid <https://github.com/geeknoid> it seems like a bug in
> "page.previous.url" if it needs this prefix, no ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/istio.github.io/pull/1035#pullrequestreview-101724765>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AVucHR5-KlCUFdWhSPka_s_BuFAvugJBks5tbwCngaJpZM4STdGK>
> .
>

vadimeisenbergibm commented on _docs/tutorial/07-canary-release-with-istio.md:
````diff
@@ -0,0 +1,43 @@
+   istioctl create -f samples/bookinfo/kube/route-rule-all-v1.yaml
+   ```
+
+2. Let's deploy our new version of the _reviews_ microservice. This time we will deploy it with the _app_ label, since Istio will route the traffic to _v1_ anyway. No traffic will arrive to our new version of the _reviews_ microservice.
````
@geeknoid 
> I don't understand the point of the app label being applied. Can you expand the description a bit of why to apply this label?

In the previous module, without Istio, the pod is added without the `app` label, so the Kubernetes service `reviews` is applied to it and no traffic is sent to it. Here, with Istio, we just define a route rule to send the traffic to version `v1`, so we can safely add version `v2` with the `app` label. The Kubernetes service `reviews` will apply to it, but the traffic will not be sent to it due to the routing rule. I will add an explanation of it.

vadimeisenbergibm commented:
@geeknoid thanks a lot for your comments! I have applied them.  rake test passes.

The plan is as follows, I will work on it next week:
1. Add more explanations to the modules about dashboard, service graph, distributed tracing
2. Add output/pictures
3. Rewrite everything to use the new route rules, so we will publish the tutorial with the new syntax

ldemailly commented:
and add tests to be able to checkin extra yamls :)

sdake commented:
@vadimeisenbergibm is the subject still accurate (hold until 0.7?)  Can you merge with master branch?  This should pick up the spell-check and MD linter gate @geeknoid implemented which should be helpful in producing better results and reducing reviewer workload :)

vadimeisenbergibm commented:
@sdake I have changed the title. I still have to do the following:
1. Complete the refactoring of the Bookinfo files
2. Complete the tutorial tests
3. Complete the tutorial itself, add output examples
4. Switch to the alpha3 route rules

sdake commented:
@vadimeisenbergibm Thanks for clarifying more work is needed before this can merge.

vadimeisenbergibm commented:
@geeknoid Thanks, I will change the tutorial accordingly.

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: 
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **linsun**

Assign the PR to them by writing `/assign @linsun` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/istio/istio.github.io/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["linsun"]} -->

linsun commented:
probably a good time to get this updated and merged around 0.8 goes out @vadimeisenbergibm :)

vadimeisenbergibm commented:
@linsun agreed, and it will test most of the features of Istio as well.

geeknoid commented:
Should we close this PR?

vadimeisenbergibm commented:
Closing until the tutorial will be updated to v1alpha3.

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio.io/pull/1035#" title="Author self-approved">vadimeisenbergibm</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **sdake**

If they are not already assigned, you can assign the PR to them by writing `/assign @sdake` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[content/OWNERS](https://github.com/istio/istio.io/blob/master/content/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["sdake"]} -->

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio.io/pull/1035#" title="Author self-approved">vadimeisenbergibm</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **sdake**

If they are not already assigned, you can assign the PR to them by writing `/assign @sdake` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[content/OWNERS](https://github.com/istio/istio.io/blob/master/content/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["sdake"]} -->

vadimeisenbergibm commented:
@sdake I will work to complete this tutorial before 1.1 release. Would be happy to hear your reviews once I will finish.

This time the tutorial will have the following features:
1.  The configurations will be specified per namespace, so multiple participants in a class could perform the tutorial on the same cluster, each participant using their own namespace.
2. It will use inline configurations, so no need to update istio/istio configurations, no need to work in a specific directory. `kubectl` can work with files from the web, so again, no need to work in a specific directory/with specific file paths any more.

Note that the feature 1 above depends on implementation of scoping in Istio 1.1 (planned). The idea is that ingress gateways/virtual services/destination rules should be specified per namespace and should not interfere one with another.

christian-posta commented:
Is this tutorial published somewhere? Or will it be going into the Istio docs?

vadimeisenbergibm commented:
@christian-posta It is going to be part of Istio docs.

christian-posta commented:
Okay awesome! The content there is looking great

On Wed, Feb 13, 2019 at 12:23 PM Vadim Eisenberg <notifications@github.com>
wrote:

> @christian-posta <https://github.com/christian-posta> It is going to be
> part of Istio docs.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/istio.io/pull/1035#issuecomment-463332122>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AADP0e-aYel0ather_RwszQPRALk_KECks5vNGYbgaJpZM4STdGK>
> .
>


-- 
*Christian Posta*
twitter: @christianposta
http://blog.christianposta.com <http://www.christianposta.com/blog>

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio.io/pull/1035#" title="Author self-approved">vadimeisenbergibm</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **louiscryan**

If they are not already assigned, you can assign the PR to them by writing `/assign @louiscryan` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/istio/istio.io/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["louiscryan"]} -->

istio-testing commented:
@vadimeisenbergibm: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

vadimeisenbergibm commented:
@sdake @christian-posta @geeknoid @rcaballeromx The tutorial is ready, tested on istio-release-1.1-20190214-09-16.

The important change comparing to the previous version is that it can be followed by multiple participants in a class simultaneously on a single cluster, each participant in their own namespace. All the modules are written "namespace-aware".

Any comments are welcome, technical and language-related. I know it is a large chunk to review, please take your time.

vadimeisenbergibm commented:
@geeknoid Several issues with this PR.

1. Do you have an icon for Tutorial?
2. I have to use bookinfo files from release-1.1, so I cannot use `github_file`.
3. I use several URLs which I ask participants to add to /etc/hosts. The URLs are my-istio-dashboard.io , my-istio-tracing.io, my-istio-logs-database.io, my-istio-servicegraph.io. I define Kubernetes Ingress with these URLs, so no port forwarding will be required. Is it possible to exclude them from link checking?

vadimeisenbergibm commented:
@geeknoid additional question - is it possible to deploy this branch, so the tutorial could be used already with a temporary deployment?

linsun commented:
@ibm-jason could you run through the tutorial and provide feedback here? Thanks.

ibm-jason commented:
@linsun I certainly can!

vadimeisenbergibm commented:
@ibm-jason Any feedback is very welcome, technical and language-wise.

rcaballeromx commented:
I'm doing a review but this one will take longer.
/hold

geeknoid commented on .spelling:
```diff
@@ -347,6 +350,7 @@ prefetching
 preformatted
 prepends
 prober
+productpage
```
This should not be in the dictionary. Put backticks around its use,.

geeknoid commented:
Happy Birthday, this PR is 1 year old today :-)

geeknoid commented:
Sorry, I hadn't seen your questions:

* Do you have an icon for Tutorial?

Let me go find something in the right style.

* I have to use bookinfo files from release-1.1, so I cannot use `github_file`.

github_file points to the release-1.1 branch. So please use it, otherwise things will get out of sync.

* I use several URLs which I ask participants to add to /etc/hosts. The URLs are my-istio-dashboard.io , my-istio-tracing.io, my-istio-logs-database.io, my-istio-servicegraph.io. I define Kubernetes Ingress with these URLs, so no port forwarding will be required. Is it possible to exclude them from link checking?

There's no good mechanism for excluding those. Can you put them in backticks?

geeknoid commented:
> additional question - is it possible to deploy this branch, so the tutorial could be used already with a temporary deployment?

The easiest thing is to use the deploy/netlify link from this PR.

geeknoid commented:
If you rebase, you can now use "classroom" as the icon name for the section.

geeknoid commented:
Deploy preview for *preliminary-istio* failed.

Built with commit a7afa59b5f6d83cc930c77342b0f5d14d39e1c4a

https://app.netlify.com/sites/preliminary-istio/deploys/5ca90b21521ec00008669002

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **louiscryan**

If they are not already assigned, you can assign the PR to them by writing `/assign @louiscryan` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/istio/istio.io/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["louiscryan"]} -->

geeknoid commented:
This PR has been around for over a year. I'm going to close it for now. Please reopen if you have time to push this over the finish line.

vadimeisenbergibm commented:
@geeknoid @rcaballeromx @frankbu I have finally completed the tutorial, see its description https://github.com/istio/istio.io/pull/1035#issue-171412522. I extended it to contain about 50% more than the original PR, added screenshots, added extensive use of Kiali. It is a large lab now, which can be taught in 6-12 hours.

I know it is a large chunk of work to preview, please take your time.

cc @sdake @christian-posta

istio-testing commented:
@vadimeisenbergibm: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

istio-testing commented:
@vadimeisenbergibm: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio.io-lint | b887226dc8519861e7b1206007c8dfe1fe6ad72b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/istio.io-lint/28) | `/test istio.io-lint`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@vadimeisenbergibm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio.io-lint | b887226dc8519861e7b1206007c8dfe1fe6ad72b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/istio.io-lint/28) | `/test istio.io-lint`
lint_istio.io | b887226dc8519861e7b1206007c8dfe1fe6ad72b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/lint_istio.io/32) | `/test lint_istio.io`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@vadimeisenbergibm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio.io-lint | b887226dc8519861e7b1206007c8dfe1fe6ad72b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/istio.io-lint/28) | `/test istio.io-lint`
lint_istio.io | b887226dc8519861e7b1206007c8dfe1fe6ad72b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/lint_istio.io/2754) | `/test lint_istio.io`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

geeknoid commented:
Keep this alive for a while...

istio-testing commented:
@vadimeisenbergibm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio.io-lint | b887226dc8519861e7b1206007c8dfe1fe6ad72b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/istio.io-lint/28) | `/test istio.io-lint`
lint_istio.io | b887226dc8519861e7b1206007c8dfe1fe6ad72b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/lint_istio.io/2759) | `/test lint_istio.io`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@vadimeisenbergibm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio.io-lint | b887226dc8519861e7b1206007c8dfe1fe6ad72b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/istio.io-lint/28) | `/test istio.io-lint`
lint_istio.io | 24e86ecd8077602a9690c565dbda6bcddec2c519 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/lint_istio.io/2775) | `/test lint_istio.io`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@vadimeisenbergibm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio.io-lint | b887226dc8519861e7b1206007c8dfe1fe6ad72b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/istio.io-lint/28) | `/test istio.io-lint`
lint_istio.io | f2de754b1067dd56a8de1f52d0b09c9b2f002970 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/lint_istio.io/2776) | `/test lint_istio.io`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@vadimeisenbergibm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio.io-lint | b887226dc8519861e7b1206007c8dfe1fe6ad72b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/istio.io-lint/28) | `/test istio.io-lint`
lint_istio.io | 1f1879762c83203f0c8fdcf271019e93dc53b471 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/lint_istio.io/2779) | `/test lint_istio.io`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@vadimeisenbergibm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio.io-lint | b887226dc8519861e7b1206007c8dfe1fe6ad72b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/istio.io-lint/28) | `/test istio.io-lint`
lint_istio.io | edbceca3180cdcec9e3834304e8e2daf6b1d23e0 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/lint_istio.io/2818) | `/test lint_istio.io`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@vadimeisenbergibm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio.io-lint | b887226dc8519861e7b1206007c8dfe1fe6ad72b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/istio.io-lint/28) | `/test istio.io-lint`
lint_istio.io | 5bebac43929775c17bcb0c136b452a348f542084 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/lint_istio.io/2893) | `/test lint_istio.io`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@vadimeisenbergibm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio.io-lint | b887226dc8519861e7b1206007c8dfe1fe6ad72b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/istio.io-lint/28) | `/test istio.io-lint`
lint_istio.io | dbc9826f56b2c9b9a9af6d6700977af3003c75da | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/lint_istio.io/2897) | `/test lint_istio.io`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@vadimeisenbergibm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio.io-lint | b887226dc8519861e7b1206007c8dfe1fe6ad72b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/istio.io-lint/28) | `/test istio.io-lint`
lint_istio.io | bebf1b11dac51c6142196a88ce8d9c4b5cef8f8a | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/lint_istio.io/2898) | `/test lint_istio.io`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@vadimeisenbergibm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio.io-lint | b887226dc8519861e7b1206007c8dfe1fe6ad72b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/istio.io-lint/28) | `/test istio.io-lint`
lint_istio.io | c0d45dbc6a736731797489e4187444a455dca930 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/lint_istio.io/2899) | `/test lint_istio.io`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@vadimeisenbergibm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio.io-lint | b887226dc8519861e7b1206007c8dfe1fe6ad72b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/istio.io-lint/28) | `/test istio.io-lint`
lint_istio.io | 4754c2cb754b799be3a1c2e4c42992eccf4033c1 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio.io/1035/lint_istio.io/2900) | `/test lint_istio.io`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->
